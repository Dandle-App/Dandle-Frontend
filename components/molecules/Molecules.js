import React from 'react';
import {View, Text, TouchableOpacity, Image} from 'react-native';
import {
  TextInputField, StyledInputLabel, RightIcon,
  LeftIcon,
} from '../atoms/Atoms';

import {colors} from '../../styles/styles';
const {background, borderColorDark, primary, secondary, textLink,
  textLight, textDark, backgroundLight} = colors;

import {Octicons} from '@expo/vector-icons';

export const StyldTextInput = ({label, icon, isPassword, hidePassword, setHidePassword, ...props}) => {
  return (
    <View>
      <LeftIcon><Octicons name={icon} color={primary} size={15}px /></LeftIcon>
      <StyledInputLabel>{label}</StyledInputLabel>
      <TextInputField {...props} />
      {isPassword && <RightIcon onPress={() => setHidePassword(!hidePassword)}>
        <Octicons name={hidePassword ? 'eye':'eye-closed'} color={primary} size={18} />
      </RightIcon>}
    </View>
  );
};

export const OrderCard = ({order}) => {
  const {customerName, customerCity, customerId, orderDate, orderStatus, orderTotal, orderItems, orderImage} = order;

  return (
    <View style={{backgroundColor: backgroundLight, borderColor: borderColorDark, borderWidth: 1,  
                  marginBottom: 10, borderRadius: 25, width: '100%'}}>
      <View style={{flexDirection: 'row', padding: 10, justifyContent: 'center', alignItems: 'center'}}>
        <Image source={{orderImage}} style={{width: '33%'}}/>
        <Text style={{width: '33%', fontWeight: 'bold', textAlign: 'center'}}>{orderStatus}</Text>
        <Text style={{width: '33%', textAlign: 'center', color: borderColorDark}}>{orderDate}</Text>
      </View>
      <Text style={{marginLeft: 20, paddingBottom: 10, color: borderColorDark}}>productNmae</Text>
    </View>
  );
};