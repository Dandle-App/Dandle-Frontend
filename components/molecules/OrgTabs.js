import * as React from 'react';
import { Image } from 'react-native';
import { NavigationContainer } from '@react-navigation/native';
import { createBottomTabNavigator } from '@react-navigation/bottom-tabs';

//screens
import OrgSettings from '../../screens/org/OrgSettings';
import OrgHomeScreen from '../../screens/org/OrgHomeScreen';
import OrgAnalytics from '../../screens/org/OrgAnalytics';
import OrgChat from '../../screens/org/OrgChat'; 

//icons
const SETTINGS = require('../../assets/settings.png');
const HOME = require('../../assets/home.png');
const CHAT = require('../../assets/chat.png');
const ANALYTICS = require('../../assets/analytics.png');

//transparent icons
const SETTINGSTRANSPARENT = require('../../assets/settingsTransparent.png');
const HOMETRANSPARENT = require('../../assets/homeTransparent.png');
const CHATTRANSPARENT = require('../../assets/chatTransparent.png');
const ANALYTICSTRANSPARENT = require('../../assets/analyticsTransparent.png');


const Tab = createBottomTabNavigator();

export default function OrgTabs() {
  return (
    <NavigationContainer>
      <Tab.Navigator
        screenOptions={{
          tabBarShowLabel: false,
          tabBarStyle: {
            position: 'absolute', 
            backgroundColor: '#f38484', 
            borderTopRightRadius: 25,
            borderTopLeftRadius: 25,},
            tabBarActiveTintColor: '#f38484'
          }
        }
      >
        <Tab.Screen 
          name="Settings" 
          component={OrgSettings}
          options={{
            tabBarIcon: ({size,focused,color}) => {
              return(
                <Image 
                  style={{width: size, height: size}}
                  //source={SETTINGS}
                  source={SETTINGSTRANSPARENT}
                />
              )
            }
          }}
        />
        <Tab.Screen 
          name="Home" 
          component={OrgHomeScreen}
          options={{
            tabBarIcon: ({size,focused,color}) => {
              return(
                <Image 
                  style={{width: size, height: size}}
                  //source={HOME}
                  source={HOMETRANSPARENT}
                />
              )
            }
          }} 
        />
        <Tab.Screen 
          name="Chat" 
          component={OrgChat} 
          options={{
            tabBarIcon: ({size,focused,color}) => {
              return(
                <Image 
                  style={{width: size, height: size}}
                  //source={CHAT}
                  source={CHATTRANSPARENT}
                />
              )
            }
          }}
        />
        <Tab.Screen 
          name="Analytics" 
          component={OrgAnalytics} 
          options={{
            tabBarIcon: ({size,focused,color}) => {
              return(
                <Image 
                  style={{width: size, height: size}}
                  //source={ANALYTICS}
                  source={ANALYTICSTRANSPARENT}
                />
              )
            }
          }}
        />
      </Tab.Navigator>
    </NavigationContainer>
  );
}