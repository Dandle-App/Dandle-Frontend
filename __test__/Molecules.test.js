import React from 'react';
import {cleanup, render, fireEvent, waitFor} from '@testing-library/react-native';
import { toHaveStyle } from "@testing-library/jest-native";
import styled from 'styled-components';
import NavigationContainer from '@react-navigation/native';
import renderer from 'react-test-renderer';
import 'jest-styled-components'
import { StyldTextInput, OrderCard } from '../components/molecules/Molecules';

afterEach(cleanup);

describe('Molecules', () => {

    it('StyledTextInput', () =>{
        const tree = renderer.create(
            <div>
            <MsgBox type="success">success</MsgBox>
            <MsgBox type="fail">fail</MsgBox>
            </div>
          ).toJSON();
          expect(tree).toMatchSnapshot();

    });
});