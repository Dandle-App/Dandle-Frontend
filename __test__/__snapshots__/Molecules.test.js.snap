// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Molecules OrderCard renders correctly 1`] = `
<div>
  <View
    style={
      Object {
        "backgroundColor": "#ffffff",
        "borderColor": "#9e9e9e",
        "borderRadius": 25,
        "borderWidth": 1,
        "marginBottom": 30,
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "padding": 10,
        }
      }
    >
      <Image
        source={
          Object {
            "uri": "https://cdn.pixabay.com/photo/2020/05/17/04/22/pizza-5179939_960_720.jpg",
          }
        }
        style={
          Object {
            "aspectRatio": 1,
            "borderRadius": 6,
            "width": "33%",
          }
        }
      />
      <Text
        style={
          Object {
            "fontFamily": "Roboto",
            "fontSize": 18,
            "fontWeight": "bold",
            "textAlign": "center",
            "width": "33%",
          }
        }
      />
      <Text
        style={
          Object {
            "color": "#bdbdbd",
            "fontSize": 12,
            "fontWeight": "bold",
            "textAlign": "right",
            "width": "33%",
          }
        }
      />
    </View>
    <Text
      style={
        Object {
          "color": "#bdbdbd",
          "fontSize": 13,
          "fontWeight": "bold",
          "marginLeft": 20,
          "paddingBottom": 10,
        }
      }
    />
  </View>
</div>
`;

exports[`Molecules StyledTextInput renders correctly 1`] = `
<div>
  <View>
    <View
      style={
        Array [
          Object {
            "left": "2%",
            "position": "absolute",
            "top": "60%",
            "zIndex": 1,
          },
        ]
      }
    >
      <Text />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#f38484",
            "fontSize": 16,
            "paddingBottom": 0,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
            "textAlign": "left",
          },
        ]
      }
    >
      Username
    </Text>
    <TextInput
      allowFontScaling={true}
      placeholder="johndoe"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "#fafafa",
            "borderBottomWidth": 3,
            "borderColor": "#d3d3d3",
            "color": "#212121",
            "marginBottom": 35,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 5,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="FakeName"
    />
  </View>
  <View>
    <View
      style={
        Array [
          Object {
            "left": "2%",
            "position": "absolute",
            "top": "60%",
            "zIndex": 1,
          },
        ]
      }
    >
      <Text />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#f38484",
            "fontSize": 16,
            "paddingBottom": 0,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
            "textAlign": "left",
          },
        ]
      }
    >
      Hidden Password
    </Text>
    <TextInput
      allowFontScaling={true}
      placeholder="* * * * * * *"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "backgroundColor": "#fafafa",
            "borderBottomWidth": 3,
            "borderColor": "#d3d3d3",
            "color": "#212121",
            "marginBottom": 35,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 5,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="fakePassword"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 35,
          "opacity": 1,
          "position": "absolute",
          "right": 1,
          "top": 35,
          "width": 35,
          "zIndex": 200,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View>
    <View
      style={
        Array [
          Object {
            "left": "2%",
            "position": "absolute",
            "top": "60%",
            "zIndex": 1,
          },
        ]
      }
    >
      <Text />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#f38484",
            "fontSize": 16,
            "paddingBottom": 0,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
            "textAlign": "left",
          },
        ]
      }
    >
      Plain Password
    </Text>
    <TextInput
      allowFontScaling={true}
      placeholder="* * * * * * *"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "backgroundColor": "#fafafa",
            "borderBottomWidth": 3,
            "borderColor": "#d3d3d3",
            "color": "#212121",
            "marginBottom": 35,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 5,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="fakePassword"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 35,
          "opacity": 1,
          "position": "absolute",
          "right": 1,
          "top": 35,
          "width": 35,
          "zIndex": 200,
        }
      }
    >
      <Text />
    </View>
  </View>
  <View>
    <View
      style={
        Array [
          Object {
            "left": "2%",
            "position": "absolute",
            "top": "60%",
            "zIndex": 1,
          },
        ]
      }
    >
      <Text />
    </View>
    <Text
      style={
        Array [
          Object {
            "color": "#f38484",
            "fontSize": 16,
            "paddingBottom": 0,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
            "textAlign": "left",
          },
        ]
      }
    >
      Hidden Password
    </Text>
    <TextInput
      allowFontScaling={true}
      placeholder="* * * * * * *"
      rejectResponderTermination={true}
      secureTextEntry={true}
      style={
        Array [
          Object {
            "backgroundColor": "#fafafa",
            "borderBottomWidth": 3,
            "borderColor": "#d3d3d3",
            "color": "#212121",
            "marginBottom": 35,
            "paddingBottom": 10,
            "paddingLeft": 10,
            "paddingRight": 10,
            "paddingTop": 5,
          },
        ]
      }
      underlineColorAndroid="transparent"
      value="fakePassword"
    />
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "height": 35,
          "opacity": 1,
          "position": "absolute",
          "right": 1,
          "top": 35,
          "width": 35,
          "zIndex": 200,
        }
      }
    >
      <Text />
    </View>
  </View>
</div>
`;
